rest_framework/utils/field_mapping.py

```diff
@@ -123,10 +123,7 @@ def get_field_kwargs(field_name, model_field):
             kwargs['allow_folders'] = model_field.allow_folders
 
     if model_field.choices:
-        # If this model field contains choices, then return early.
-        # Further keyword arguments are not valid.
         kwargs['choices'] = model_field.choices
-        return kwargs
```

Comment:
Granted this may not be failing any tests, but I'm pretty wary.

Let's make sure that we're testing the case that this describes (a field that would result in adding keyword args that *aren't* valid on ChoiceField)